test/create-logical-locks/logicallocks.mlir

```diff
@@ -0,0 +1,91 @@
+//===- locks1.mlir ---------------------------------------------*- MLIR -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2021 Xilinx Inc.
+//
+//===----------------------------------------------------------------------===//
+
+// RUN: aie-opt --aie-create-logical-lock %s | FileCheck %s
+
+// CHECK:  module @test_xaie0 {
+// CHECK:    %0 = AIE.tile(1, 1)
+// CHECK:    %1 = AIE.tile(3, 4)
+// CHECK:    %2 = AIE.tile(3, 2)
+// CHECK:    %3 = AIE.tile(3, 3)
+// CHECK:    %4 = AIE.tile(4, 3)
+// CHECK:    %5 = AIE.lock(%0, 0)
+// CHECK:    %6 = AIE.lock(%3, 0)
+// CHECK:    %7 = AIE.lock(%3, 1)
+// CHECK:    %8 = AIE.lock(%4, 0)
+// CHECK:    %9 = AIE.core(%0) {
+// CHECK:      AIE.useLock(%5, Acquire, 0)
+// CHECK:      AIE.useLock(%5, Release, 1)
+// CHECK:      AIE.end
+// CHECK:    }
```

Comment:
For the purposes of this test, the useLock operations are irrelevent, I think?  If so, it's usually good practice to reduce the size of the test to the absolute minimum.  Also, there are some behaviors you haven't tested (like what happens when there are already lock ids).